File: elastic-consumer-node/src/queues/createConnection.ts
import {Connection} from 'amqplib'
import amqp from 'amqplib'
import { elasticLogger } from '../libs/common.logger'
import { AMQPException } from '../exceptions/index'
import { IConnectionConfig } from './types'


async function createConnection(amqpOptions : IConnectionConfig) : Promise<any> {

    let retryConnectionCount = 4
    let retryConnectionStatus = true

    const connectRabbitMq = async () : Promise<any | Connection>  => {
        try{    
            const {url : rabbitmqUrl} = amqpOptions

            if(!rabbitmqUrl){
                throw new AMQPException(`There is no any Rabbitmq Url , Please Construct the valid Rabbitmq Url`)
            }

            const connection =  amqp.connect(rabbitmqUrl)
            return connection
        }catch(err){
            console.log(err)
            if(retryConnectionCount.toString().startsWith('0')) {
                elasticLogger.error(`The Maximum Retry Connection Count is Exceeded, Try again Later.. Retry Count : ${retryConnectionCount}`)
                return null
            }
            retryConnectionCount -= 1
            elasticLogger.error(`Retrying the RabbitMQ Connection Retry Count : ${retryConnectionCount}`)
            connectRabbitMq()
        }
    }

    while(retryConnectionStatus && !retryConnectionCount.toString().startsWith('0')) {
        const result : Connection = await connectRabbitMq()
        if(!result){
            continue
        }
        return result
    }
}

export {
    createConnection
}

File: rabbitmq-reviewer-python/queues/CreateConnection.py
import pika
from pika.exceptions import AMQPConnectionError
from config.EnvConfig import env_config


def create_connection():

    retry_connection_status = True
    retry_connection_count = 4

    while retry_connection_status and retry_connection_count > 0:

        try:
            url = env_config['url']
            connection = pika.BlockingConnection(pika.ConnectionParameters(url))
            return connection
        except AMQPConnectionError as amqp_error:
            is_maximum_exceeded = str(retry_connection_count).startswith('0')
            if is_maximum_exceeded:
                retry_connection_status = False
                print(f'Maximum Exceeded Retry Count , Error Trying to RabbitMQ Connection Failed : {amqp_error}')

            print(f'Error Connecting to the Rabbit MQ, Trying again Retry Count : {retry_connection_count}')
            retry_connection_status -= 1
            create_connection()
